hazelcast/src/main/java/com/hazelcast/map/impl/wan/WanMapEntryView.java

```diff
@@ -0,0 +1,252 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.wan;
+
+import com.hazelcast.core.EntryView;
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.nio.IOUtil;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.BinaryInterface;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+
+/**
+ * WAN heap based implementation of {@link EntryView} for keeping
+ * compatibility when sending to older (3.8+) clusters.
+ *
+ * @param <K> the type of key.
+ * @param <V> the type of value.
+ */
+@BinaryInterface
+public class WanMapEntryView<K, V> implements EntryView<K, V>, IdentifiedDataSerializable {
+
+    private final K key;
+    private final V value;
+    private final long cost;
+    private final long creationTime;
+    private final long expirationTime;
+    private final long hits;
+    private final long lastAccessTime;
+    private final long lastStoredTime;
+    private final long lastUpdateTime;
+    private final long version;
+    private final long ttl;
+
+    public WanMapEntryView(EntryView<K, V> entryView) {
+        this.key = entryView.getKey();
+        this.value = entryView.getValue();
+        this.cost = entryView.getCost();
+        this.version = entryView.getVersion();
+        this.hits = entryView.getHits();
+        this.lastAccessTime = entryView.getLastAccessTime();
+        this.lastUpdateTime = entryView.getLastUpdateTime();
+        this.ttl = entryView.getTtl();
+        this.creationTime = entryView.getCreationTime();
+        this.expirationTime = entryView.getExpirationTime();
+        this.lastStoredTime = entryView.getLastStoredTime();
+    }
+
+    @Override
+    public K getKey() {
+        return key;
+    }
+
+    @Override
+    public V getValue() {
+        return value;
+    }
+
+    @Override
+    public long getCost() {
+        return cost;
+    }
+
+    @Override
+    public long getCreationTime() {
+        return creationTime;
+    }
+
+    @Override
+    public long getExpirationTime() {
+        return expirationTime;
+    }
+
+    @Override
+    public long getHits() {
+        return hits;
+    }
+
+    @Override
+    public long getLastAccessTime() {
+        return lastAccessTime;
+    }
+
+    @Override
+    public long getLastStoredTime() {
+        return lastStoredTime;
+    }
+
+    @Override
+    public long getLastUpdateTime() {
+        return lastUpdateTime;
+    }
+
+    @Override
+    public long getVersion() {
+        return version;
+    }
+
+    @Override
+    public long getTtl() {
+        return ttl;
+    }
+
+    @Override
+    public long getMaxIdle() {
+        return 0;
+    }
+
+    /**
+     * Needed for client protocol compatibility.
+     */
+    @SuppressWarnings("unused")
+    public long getEvictionCriteriaNumber() {
```

Comment:
Why keep the `evictionCriteriaNumber` methods since they are unused & `WanMapEntryView` is only meant for WAN usage?

Reply:
Copy-paste, removed.